src/pypromice/process/aws.py

```diff
@@ -38,84 +41,82 @@ def __init__(self, config_file, inpath, var_file=None, meta_file=None):
             variables file is used. The default is None.
         meta_file: str, optional
-            Metadata info file path. If not given then pypromice's 
+            Metadata info file path. If not given then pypromice's
             metadata file is used. The default is None.
         '''
         assert(os.path.isfile(config_file)), "cannot find "+config_file
         assert(os.path.isdir(inpath)), "cannot find "+inpath
-        print('\nAWS object initialising...')
-        
+        logger.info('AWS object initialising...')
+
         # Load config, variables CSF standards, and L0 files
         self.config = self.loadConfig(config_file, inpath)
         self.vars = getVars(var_file)
         self.meta = getMeta(meta_file)
 
-        # Hard-wire the msg_lat and msg_lon here
-        # Prevents having to list these vars in the individual station toml files
-        config_keys = list(self.config.keys())
-        for i in config_keys:
-            self.config[i]['columns'].extend(['msg_lat', 'msg_lon'])
-
         # Load config file
         L0 = self.loadL0()
         self.L0=[]
         for l in L0:
             n = getColNames(self.vars, l.attrs['number_of_booms'], l.attrs['format'])
             self.L0.append(popCols(l, n))
-      
+
+        self.L1 = None
+        self.L1A = None
+        self.L2 = None
+        self.L3 = None
+
     def process(self):
         '''Perform L0 to L3 data processing'''
         try:
-            print(f'Commencing {self.L0.attrs["number_of_booms"]}-boom processing...')
+            logger.info(f'Commencing {self.L0.attrs["number_of_booms"]}-boom processing...')
         except:
-            print(f'Commencing {self.L0[0].attrs["number_of_booms"]}-boom processing...')        
+            logger.info(f'Commencing {self.L0[0].attrs["number_of_booms"]}-boom processing...')
         self.getL1()
         self.getL2()
         self.getL3()
-        
+
     def write(self, outpath):
         '''Write L3 data to .csv and .nc file'''
         if os.path.isdir(outpath):
             self.writeArr(outpath)
         else:
-            print(f'Outpath f{outpath} does not exist. Unable to save to file')
+            logger.info(f'Outpath f{outpath} does not exist. Unable to save to file')
             pass
-            
+
     def getL1(self):
         '''Perform L0 to L1 data processing'''
-        print('Level 1 processing...')
+        logger.info('Level 1 processing...')
         self.L0 = [addBasicMeta(item, self.vars) for item in self.L0]
         self.L1 = [toL1(item, self.vars) for item in self.L0]
-        self.L1A = mergeVars(self.L1, self.vars)
+        self.L1A = reduce(xr.Dataset.combine_first, self.L1)
 
     def getL2(self):
         '''Perform L1 to L2 data processing'''
-        print('Level 2 processing...')
-        self.L2 = toL2(self.L1A)
-        self.L2 = clipValues(self.L2, self.vars)
+        logger.info('Level 2 processing...')
+        self.L2 = toL2(self.L1A, vars_df=self.vars)
```

Comment:
Bug found here. `vars_df` is not an optional variable now, so should be changed to:
```
self.L2 = toL2(self.L1A, self.vars)
```